### ISSUE TITLE
[4.x]: Can't create entry on non-default site

### ISSUE THREAD

**Issue by tremby** [craftcms/cms#10902]
### Description

I have two sites. I can't seem to create an entry on the non-default site.

### Steps to reproduce

1. Have a multisite Craft
2. Go to entries page
3. See that I have multiple pages on default site
4. Switch to non-default site with the dropdown
5. See that I have no pages on this site
6. Click "new entry"
7. Fill in some things and save

### Expected behavior

New entry is associated with the non-default site I had chosen.

### Actual behavior

New entry is associated with the default site.

I'm somewhat new to Craft so I may be missing something obvious. But I can't see any options when making an entry to choose which site it's part of, so my assumption was that switching site on the entries list before clicking "new entry" was the way to do it.

### Craft CMS version

4.0.x-dev at 6e466a0

### PHP version

8.1.2

### Operating system and version

Linux 5.13.0

### Database type and version

Mysql 8.0.27

### Installed plugins and versions

- Control Panel CSS v2.x-dev
- Formie dev-craft-4
- Navigation dev-craft-4
- Redactor 3.0.0-beta.2
- Retour 4.0.0-beta.4
- SEOmatic 4.0.0-beta.7
- Typed link field 2.1.0-rc
- Vizy dev-craft-4
- Workflow dev-craft-4

**Comment by tremby** [comment]
I checked the network panel on clicking "new entry". The request which goes out indicates the non-default site, with site ID 2.

`{"elementType":"craft\\elements\\Entry","siteId":2,"sectionId":1,"typeId":1,"enabled":1}`

But in the response is `cpEditUrl: ".../admin/entries/pages/73?site=default&draftId=40"` and that's the URL I end up on, and `site=default` does not sound right.

Interestingly, after the initial POST which was successful, I then see a GET query for the URL which was returned in the response, and the response is a 500. The response body doesn't look like an error message, however. But then I tried this again on the default site and see the same 500 as a response to the GET after the initial POST.
